crates/outbound-pg/src/lib.rs

```diff
@@ -325,3 +344,47 @@ impl std::fmt::Debug for PgNull {
         f.debug_struct("NULL").finish()
     }
 }
+
+/// Delegate a function call to the v2::HostConnection implementation
+macro_rules! delegate {
+    ($self:ident.$name:ident($address:expr, $($arg:expr),*)) => {{
+        let connection = match <Self as v2::HostConnection>::open($self, $address).await? {
+            Ok(c) => c,
+            Err(e) => return Ok(Err(to_legacy_error(e))),
+        };
+        Ok(<Self as v2::HostConnection>::$name($self, connection, $($arg),*)
+            .await?
+            .map_err(|e| to_legacy_error(e)))
+    }};
+}
```

Comment:
I'm on the fence with these macros. It's a nice pattern when you're _writing_ the delegation, but I'm not sure its worth it for the next person who needs to maintain these (even if thats just you in 6 months).

Reply:
Perhaps I've written too many macros, but I personally find this one easy to understand. I also find it easier to understand the intent of the code which is that it really is meant to mechanically delegate. 

Perhaps we leave this in and see how the age? Inlining them should be easy to do if folks find them hard to understand in the future.